spago.yaml

```
@@ -0,0 +1,39 @@
+package:
+  name: profunctor-lenses
+  publish:
+    license: MIT
+    version: 9.0.0
+    location:
+      githubOwner: purescript-contrib
+      githubRepo: purescript-profunctor-lenses
+  dependencies:
+    - arrays: ">=7.3.0 <8.0.0"
+    - bifunctors: ">=6.0.0 <7.0.0"
+    - const: ">=6.0.0 <7.0.0"
+    - control: ">=6.0.0 <7.0.0"
+    - distributive: ">=6.0.0 <7.0.0"
+    - either: ">=6.1.0 <7.0.0"
+    - foldable-traversable: ">=6.0.0 <7.0.0"
+    - foreign-object: ">=4.1.0 <5.0.0"
+    - functors: ">=5.0.0 <6.0.0"
+    - identity: ">=6.0.0 <7.0.0"
+    - lists: ">=7.0.0 <8.0.0"
+    - maybe: ">=6.0.0 <7.0.0"
+    - newtype: ">=5.0.0 <6.0.0"
+    - ordered-collections: ">=3.2.0 <4.0.0"
+    - partial: ">=4.0.0 <5.0.0"
+    - prelude: ">=6.0.1 <7.0.0"
+    - profunctor: ">=6.0.1 <7.0.0"
+    - record: ">=4.0.0 <5.0.0"
+    - safe-coerce: ">=2.0.0 <3.0.0"
+    - transformers: ">=6.1.0 <7.0.0"
+    - tuples: ">=7.0.0 <8.0.0"
+  test:
+    main: Test.Main
+    dependencies:
+      - assert
+      - effect
+workspace:
+  extraPackages: {}
+  packageSet:
+    registry: 60.5.0
```

Libraries should not use package sets at all:
```suggestion
workspace: {}
```

---

I see some errors

```sh
Reading Spago workspace configuration...

✓ Selecting package to build: affjax-node


✘ Could not solve the package dependencies, errors:
  While solving affjax each version could not be solved:
- 13.0.0:
  Conflict in version ranges for aff:
    >=8.0.0 (declared dependency)
    <8.0.0 seen in affjax@13.0.0

✓ Selecting package to build: aff-bus


✘ Could not solve the package dependencies, errors:
  While solving avar each version could not be solved:
- 5.0.0:
  Conflict in version ranges for aff:
    >=8.0.0 (declared dependency)
    <8.0.0 seen in avar@5.0.0

✓ Selecting package to build: optparse


✘ Could not solve the package dependencies, errors:
  No versions found in the registry for optparse in range
  >=5.0.1 seen in spec-node@0.0.1, spec-node@0.0.2, spec-node@0.0.3
  <6.0.0 seen in spec-node@0.0.1, spec-node@0.0.2, spec-node@0.0.3



✓ Selecting package to build: github-actions-toolkit


✘ Could not solve the package dependencies, errors:
  While solving aff-promise each version could not be solved:
- 4.0.0:
  Conflict in version ranges for aff:
    >=8.0.0 (declared dependency)
    <8.0.0 seen in aff-promise@4.0.0

```

---

That's weird. I deleted the existing lockfile, added version ranges to the dependencies, and it appears to be working for me. Here's the diff:

```diff
diff --git a/spago.yaml b/spago.yaml
index 571b4f5..2055703 100644
--- a/spago.yaml
+++ b/spago.yaml
@@ -31,6 +31,7 @@ package:
   test:
     main: Test.Main
     dependencies:
-      - assert
-      - effect
+      - assert: ">=6.0.0 <7.0.0"
+      - effect: ">=4.0.0 <5.0.0"
+
 workspace: {}
```

and of the resulting lockfile change:

```diff
diff --git a/spago.lock b/spago.lock
index 2b03085..208e19d 100644
--- a/spago.lock
+++ b/spago.lock
@@ -112,8 +112,12 @@
         },
         "test": {
           "dependencies": [
-            "assert",
-            "effect"
+            {
+              "assert": ">=6.0.0 <7.0.0"
+            },
+            {
+              "effect": ">=4.0.0 <5.0.0"
+            }
           ],
           "build_plan": [
             "assert",
```

---

Oh, I see you already were able to fix it.